geode-core/src/main/java/org/apache/geode/internal/cache/TXState.java

```diff
@@ -591,13 +593,14 @@ private void lockFilterRegistrationOnTxRegions() {
     for (InternalRegion region : getRegions()) {
       if (lockableRegionForFilterRegistration(region)) {
         region.getFilterProfile().lockFilterRegistrationDuringTx();
+        filterLockAcquiredRegions.add(region);
       }
     }
   }
 
   private void unlockFilterRegistrationOnTxRegions() {
-    for (InternalRegion region : getRegions()) {
-      if (lockableRegionForFilterRegistration(region)) {
+    for (InternalRegion region : filterLockAcquiredRegions) {
+      if (region != null) {
```

Comment:
Will this region be destroyed; do we need check for destroyed regions...

Reply:
Even if region is destroyed, we still need to unlock the readLock it holds to avoid deadlock.